tests/controllers/villains.test.js

```diff
@@ -0,0 +1,108 @@
+
+    stubbedFindOne = sandbox.stub(models.villains, 'findOne')
+    stubbedFindAll = sandbox.stub(models.villains, 'findAll')
+
+    stubbedSend = sandbox.stub()
+    stubbedSendStatus = sandbox.stub()
+    stubbedStatusSend = sandbox.stub()
+    stubbedStatus = sandbox.stub()
+
+    response = {
+      send: stubbedSend,
+      sendStatus: stubbedSendStatus,
+      status: stubbedStatus,
+    }
+  })
+
+  beforeEach(() => {
+    stubbedStatus.returns({ send: stubbedStatusSend })
+  })
+
+  afterEach(() => {
+    sandbox.reset()
+  })
+
+
+  describe('getAllVillains', () => {
+    it('retrieves a list of villains from a database and calls response.send() with the list', async () => {
+      stubbedFindAll.returns(villainsList)
+
+      await getAllVillains({}, response)
+
+      expect(stubbedFindAll).to.have.callCount(1)
+      expect(stubbedSend).to.have.been.calledWith(villainsList)
+    })
+  })
+
+  describe('getBySlug', () => {
+    it('retrieves the villain associated with the provided slug from the database and calls response.send with it', async () => {
+      stubbedFindOne.returns(singleVillain)
+      const request = { params: { slug: 'gaston' } }
+
+
+      await getBySlug(request, response)
+
+      expect(stubbedFindOne).to.have.been.calledWith({ where: { slug: 'gaston' }, attributes: ['name', 'movie', 'slug'] })
+      expect(stubbedStatusSend).to.have.been.calledWith(singleVillain)
+    })
+
+    it('returns a 404 when no villain with the matching slug is found', async () => {
+      stubbedFindOne.returns(null)
+      const request = { params: { slug: 'not-found' } }
+
+      await getBySlug(request, response)
+
+      expect(stubbedFindOne).to.have.been.calledWith({ where: { slug: 'not-found' }, attributes: ['name', 'movie', 'slug'] })
+      expect(stubbedSendStatus).to.have.been.calledWith(404)
+    })
+
+    it('returns a 500 with an error message when the database call throws an error', async () => {
+      stubbedFindOne.throws('error')
+      const request = { params: { slug: 'throw-error' } }
+
+      await getBySlug(request, response)
+
+      expect(stubbedFindOne).to.have.been.calledWith({ where: { slug: 'throw-error' }, attributes: ['name', 'movie', 'slug'] })
+      expect(stubbedStatus).to.have.been.calledWith(500)
+      expect(stubbedStatusSend).to.have.been.calledWith('Unable to retrieve villain, please try again')
+    })
+  })
+
+  describe('saveNewVillain', () => {
+    it('accepts new villain details and saves them as a new villain, returning the saved record with a 201 status', async () => {
+      const request = { body: singleVillain }
+      const stubbedCreate = sinon.stub(models.villains, 'create').returns(singleVillain)
+
+      await saveNewVillain(request, response)
+
+      expect(stubbedCreate).to.have.been.calledWith(singleVillain)
+      expect(stubbedStatus).to.have.been.calledWith(201)
+      expect(stubbedStatusSend).to.have.been.calledWith(singleVillain)
+    })
+  })
```

Comment:
Should have error handling and a test for it. Should also have a test for the 400 state where not all the data was passed in.